plugins/module_utils/CertACLHelper.cs

```diff
@@ -0,0 +1,282 @@
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using System.Security.AccessControl;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+using System.Security.Principal;
+
+
+//TypeAccelerator -Name Ansible.Windows.CertAclHelper.CryptHandle -TypeName CryptHandle
+//TypeAccelerator -Name Ansible.Windows.CertAclHelper.SafeSecurityDescriptorPtr -TypeName SafeSecurityDescriptorPtr
+//TypeAccelerator -Name Ansible.Windows.CertAclHelper.CertAccessRights -TypeName CertAccessRights
+//TypeAccelerator -Name Ansible.Windows.CertAclHelper.CertAccessRightsHelper -TypeName CertAccessRightsHelper
+//TypeAccelerator -Name Ansible.Windows.CertAclHelper.CertAclHelper -TypeName CertAclHelper
+
+namespace ansible_collections.ansible.windows.plugins.module_utils.CertACLHelper
+{
+
+    public class CryptHandle : SafeHandleZeroOrMinusOneIsInvalid
+    {
+
+        public CryptHandle()
+            : base(true)
+        {
+        }
+
+        public CryptHandle(IntPtr handle)
+            : base(true)
+        {
+            this.SetHandle(handle);
+        }
+
+        [DllImport("advapi32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool CryptReleaseContext(IntPtr safeProvHandle, uint dwFlags);
+
+        protected override bool ReleaseHandle()
+        {
+            return CryptReleaseContext(this.handle, 0);
+        }
+    }
+    public class SafeSecurityDescriptorPtr : SafeHandleZeroOrMinusOneIsInvalid
+    {
+        private int size = -1;
+
+        public SafeSecurityDescriptorPtr()
+            : base(true)
+        {
+        }
+
+        public SafeSecurityDescriptorPtr(uint size)
+            : base(true)
+        {
+            this.size = (int)size;
+            this.SetHandle(Marshal.AllocHGlobal(this.size));
+        }
+
+        public SafeSecurityDescriptorPtr(IntPtr handle)
+            : base(true)
+        {
+            this.SetHandle(handle);
+        }
+
+        protected override bool ReleaseHandle()
+        {
+            try
+            {
+                Marshal.FreeHGlobal(this.handle);
+                return true;
+            }
+            catch
+            {
+                // semantics of this function are to never throw an exception so we must eat the underlying error and 
+                // return false. 
+                return false;
+            }
+        }
+    }
+    [Flags]
+    public enum CertAccessRights : uint
+    {
+        Read = 0x80000000,
+        FullControl = 0x10000000
+    }
+
+    public static class CertAccessRightsHelper
```

Comment:
It this necessary? Seems like we can just do `[int]$accessRights` in the PowerShell code and avoid having this altogether. Potentially you don't even need the cast in PowerShell as well but you would have to test that.

Reply:
I believe this was an issue on older powershell versions, but my memory may be wrong.